pkg/cmd/printer.go

```diff
@@ -24,7 +25,49 @@ func NewPrinter(out io.Writer, wide bool) *Printer {
 	}
 }
 
-// PrintChecks prints permission checks returned by Check.
+// Struct to hold either rb or crb objects
+type rowData struct {
+	Name     string         `json:"name"`
+	RoleRef  rbac.RoleRef   `json:"roleRef" protobuf:"bytes,3,opt,name=roleRef"`
+	Subjects []rbac.Subject `json:"subjects,omitempty" protobuf:"bytes,2,rep,name=subjects"`
+}
+
+// ExportData exports data to a file.
+func (p *Printer) ExportData(action Action, roleBindings []rbac.RoleBinding, clusterRoleBindings []rbac.ClusterRoleBinding) {
+	// Final data to be exported as JSON
+	data := make(map[string]interface{})
+
+	if action.Resource != "" {
+		// NonResourceURL permissions can only be granted through ClusterRoles. Hence no point in printing RoleBindings section.
+		if len(roleBindings) != 0 {
+			rbData := []rowData{}
+			// Get required data from each roleBinding
+			for _, rb := range roleBindings {
+				rbData = append(rbData, rowData{rb.Name, rb.RoleRef, rb.Subjects})
+			}
+			data["roleBindings"] = rbData
+		}
+	}
+
+	if len(clusterRoleBindings) != 0 {
+		crbData := []rowData{}
+		// Get required data from each roleBinding
+		for _, crb := range clusterRoleBindings {
+			crbData = append(crbData, rowData{crb.Name, crb.RoleRef, crb.Subjects})
```

Comment:
I think we don't want to append ClusterRoleBindings that do not have any associated Subject. Check the logic in PrintCheck method, where we have inner loop to iterate over Subjects.

Reply:
Thanks for the review @danielpacak! I've added checks like the following to skip roleBindings and clusterRoleBindings which have no subject:
```go
if len(rb.Subjects) != 0 {
  rbData = append(rbData, rowData{rb.Name, rb.RoleRef, rb.Subjects})
}
```